### PR TITLE
feat(tsingmicro): enable vllm 0.24.0 on txda (complement #447)

### LINKED ISSUE #507: feat(tsingmicro): enable vllm 0.24.0 on txda (complement #447)
Release tracking issue for the `flagos2.2-rc2` release.

Linked PR: https://github.com/flagos-ai/vllm-plugin-FL/pull/460

[#447](https://github.com/flagos-ai/vllm-plugin-FL/pull/447) landed the txda vendor backend skeleton but it is not independently serveable on tsingmicro (verified 2026-09-08 on tsm): stock main crashes at init (no `txda→tccl` dist backend → falls to nccl on CPU-only torch) and at warmup (`torch.Generator(device=txda)` in the dummy sampler falls through flag_gems' cpu fallback and dies on device mismatch). This PR adds only the enablement strictly required to start and serve correctly on txda.

### PR DESCRIPTION
**Pull request by tengqm**
Fixes #507

## Why

[#447](https://github.com/flagos-ai/vllm-plugin-FL/pull/447) landed the txda vendor backend skeleton but it is not independently serveable on tsingmicro (verified 2026-09-08 on tsm): stock main crashes at init (no `txda→tccl` dist backend → falls to nccl on CPU-only torch) and at warmup (`torch.Generator(device=txda)` in the dummy sampler falls through flag_gems' cpu fallback and dies on device mismatch). This PR adds only the enablement strictly required to start and serve correctly on txda.

## What

- **platform**: `txda→tccl` dist backend (CPU torch has no NCCL)
- **model_runner**: txda uses a **cpu** `torch.Generator` (both the dummy-sampler and the seeded-request sites)
- **dispatch/config/tsingmicro.yaml**: attention→vendor (TxdaSDPA), rms_norm/rotary_embedding reference-first (flag_gems kernels silently compute wrong values on TX8110 — rms_norm maxrel ~7200), silu_and_mul flagos, copy-family blacklist
- **vendor/txda** (`txda.py` + `impl/attention.py`): `TxdaSDPAAttentionBackend` — reuses the flag_gems metadata machinery (KV layout, block table, slot mapping) but computes attention with torch SDPA, numerically correct on TX8110 where `flag_gems.flash_attn_varlen_func` is silently wrong
- **register_ops**: drop stale fused-moe/rms_norm registrations that referenced methods removed with the txda.py rework (txda.py now exposes only `attention_backend`; the stale OpImpls aborted the whole vendor registration)

## Verification (final commit, on-node)

tsm node, `flagos-runtime-tsingmicro-tsm260610:2.1.2`, vllm `0.24.0+flagos`, Qwen3-4B, §15 recipe:

| Path | init | 5 prompts |
|---|---|---|
| F (flagtree 0.6.1+tsm3.3) | ~698s | 5/5 correct, matches §15 verbatim |
| T (vendor triton 3.6) | ~362s | 5/5 identical to F |

Dispatch identical on both: attention→vendor.txda, rms_norm/rotary→reference.torch, silu_and_mul→flagos. Slow generation (~0.04-0.05 tok/s) is the known flag_gems index_put CPU-fallback path, not a defect.

This PR was written in part with the assistance of generative AI.

### CODE REVIEW

File: vllm_fl/worker/model_runner.py [hunk 1]
```diff
@@ -1267,7 +1267,9 @@ def _update_states(self, scheduler_output: "SchedulerOutput") -> Callable | None
                 sampling_params
                 and sampling_params.sampling_type == SamplingType.RANDOM_SEED
             ):
-                generator = torch.Generator(device=self.device)
+                generator = torch.Generator(
```
Comment (ceci3) [inline vllm_fl/worker/model_runner.py:1270]: Is the operator slow because the FlagGems operator was substituted in, or is it inherently slow?

File: vllm_fl/dispatch/backends/vendor/txda/txda.py [hunk 2]
```diff
@@ -34,97 +36,44 @@ def vendor(self) -> Optional[str]:
...
-            compute_type,
-            use_fp8_w8a8,
-            use_int8_w8a8,
-            use_int8_w8a16,
-            use_int4_w4a16,
-            per_channel_quant,
-            block_shape=block_shape,
-            B_bias=B_bias,
-        )
+    # ==================== Operator Implementations ====================
```
Comment (ceci3) [inline vllm_fl/dispatch/backends/vendor/txda/txda.py:54]: why delete fused op register on txda?

File: .gitignore [hunk 3]
```diff
@@ -14,3 +14,4 @@ coverage.xml
 # Reports
 report*.json
 test-results-*
+vllm_fl/.DS_Store
```
Comment (ceci3) [inline .gitignore:17]: why need add vllm_fl/.DS_Store

File: vllm_fl/worker/model_runner.py (hunk 1 above)
Comment (tengqm) [reply vllm_fl/worker/model_runner.py:1314]: Not the substitution — it is the device. The slow op is the index write into the paged KV cache: TX8110 has no PrivateUse1 kernel for it, so it stages through host memory, and the cost is a fixed per-call round trip rather than arithmetic — writing 1 row and writing 2048 rows both take ~166 ms on this card (measured on the 0.20.2 line, same hardware). Implementation choice moves it the other way: `rms_norm` / `rotary_embedding` on the reference path are correct *and* cheaper than the flag_gems kernels here.

File: vllm_fl/dispatch/backends/vendor/txda/txda.py (hunk 2 above)
Comment (tengqm) [reply vllm_fl/dispatch/backends/vendor/txda/txda.py:54]: Both entries pointed at methods whose whole body was `from .impl.normalization import ...` / `from .impl.fused_moe import ...`, and neither module has ever existed under `vendor/txda/` — so the two ops only ever raised ImportError.

After the rework the backend exposes `attention_backend` alone, so keeping the entries would raise AttributeError while building the list in `register_builtins`, which `_register_vendor_backends` catches and logs at debug level — silently dropping the txda attention registration with them. `rms_norm` is served by `reference` via `tsingmicro.yaml`.

Also fixed the leftover `METAX` docstring in that function (`b35565a`).

File: .gitignore (hunk 3 above)
Comment (tengqm) [reply .gitignore]: Stray — a local artifact of mine, and nothing in the tree tracks a `.DS_Store`, so the entry was doing no work. Removed in `b35565a`.

File: vllm_fl/dispatch/backends/vendor/txda/impl/attention.py [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+import torch
+
+from vllm.v1.attention.backend import AttentionType
+from vllm_fl.dispatch.backends.flaggems.impl.attention import (
+    AttentionFLBackend,
+    AttentionFLImpl,
+)
+
+_DEBUG = os.environ.get("FL_DEBUG_TXDA_ATTN") == "1"
+_PRINTED = [0]
```
Comment (ceci3) [inline vllm_fl/dispatch/backends/vendor/txda/impl/attention.py]: delete debug log and don't import vendor-spec env

Comment (tengqm) [reply vllm_fl/dispatch/backends/vendor/txda/impl/attention.py]: Done in 0a14ad1: `FL_DEBUG_TXDA_ATTN`, its `_PRINTED` counter and both `print` sites are gone, along with the now-unused `import os`.

**Review by ceci3 — APPROVED** [review]
